translator/trace/protospan_to_spandata.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Package trace defines translators from Trace proto spans to OpenCensus Go spanData
+*/
+package tracetranslator
+
+import (
+	"errors"
+	"time"
+
+	"go.opencensus.io/trace"
+	"go.opencensus.io/trace/tracestate"
+
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/golang/protobuf/ptypes/wrappers"
+)
+
+var errNilSpan = errors.New("expected a non-nil span")
+
+func ProtoSpanToOCSpanData(span *tracepb.Span) (*trace.SpanData, error) {
+	if span == nil {
+		return nil, errNilSpan
+	}
+
+	sTracestate, err := protoTracestateToOCTracestate(span.Tracestate)
+	if err != nil {
+		return nil, err
+	}
+
+	sc := trace.SpanContext{Tracestate: sTracestate}
+	copy(sc.TraceID[:], span.TraceId)
+	copy(sc.SpanID[:], span.SpanId)
+	var parentSpanID trace.SpanID
```

Comment:
trace.SpanID is a non-pointer type

Reply:
Roger that!

Reply:
Actually `trace.SpanID` is a non-nil type and is `[8]byte` so cannot be nil, but it can be zeroed. Invoking `copy(dest, []byte(nil))` won't copy anything so nothing to change :)